src/Platform/Microsoft.Testing.Platform/Hosts/CommonTestHost.cs

```diff
@@ -61,6 +92,25 @@ public async Task<int> RunAsync()
         return exitCode;
     }
 
+    private async Task<bool> IsDotnetTestHandshakeSuccessfulAsync(NamedPipeClient dotnetTestPipeClient, ITestApplicationModuleInfo? testApplicationModuleInfo, CancellationToken testApplicationCancellationToken)
+    {
+        HandshakeInfo handshakeInfo = new(new Dictionary<string, string>()
+        {
+            { HandshakeInfoPropertyNames.PID, ServiceProvider.GetProcessHandler().GetCurrentProcess().Id.ToString(CultureInfo.InvariantCulture) },
+            { HandshakeInfoPropertyNames.Architecture, RuntimeInformation.OSArchitecture.ToString() },
+            { HandshakeInfoPropertyNames.Framework, RuntimeInformation.FrameworkDescription },
+            { HandshakeInfoPropertyNames.OS, RuntimeInformation.OSDescription },
+            { HandshakeInfoPropertyNames.ProtocolVersion, ProtocolConstants.Version },
+            { HandshakeInfoPropertyNames.HostType, this is ConsoleTestHost ? nameof(ConsoleTestHost) : nameof(TestHostControllersTestHost) },
```

Comment:
This doesn't take into account potential other sub-classes. Maybe `this.GetType().Name` would be better.